annotations/builder/src/main/java/io/sundr/builder/internal/functions/ClazzAs.java

```diff
@@ -92,8 +89,11 @@ public TypeDef apply(TypeDef item) {
                 boolean isSet = TypeUtils.isSet(toAdd.getTypeRef());
                 boolean isList = TypeUtils.isList(toAdd.getTypeRef());
                 boolean isMap = TypeUtils.isMap(toAdd.getTypeRef());
-                boolean isCollection = isSet || isList;
                 boolean isAbstract = isAbstract(unwrapped);
+                boolean isOptionalInt = TypeUtils.isOptionalInt(toAdd.getTypeRef());
```

Comment:
Why do we care about the generic type? 
It seems to me that we just need to know if type is Optional, no?

Reply:
`OptionalInt` is not just an optional... They require different handling: `OptionalInt.empty()`, `OptionalInt.of(1)`, etc. See InitEnricher and BuilderUtils (although I'm not sure those init statements are used anywhere any more).

Reply:
Sure, but we don't seem to handle it here.

Reply:
Ah OK - will refactor that out.

Reply:
Updated.